packages/protocol/test/common/onboarding.ts

```diff
@@ -0,0 +1,424 @@
+        const requestTx = await relayermtw.executeTransactions(
+          transactions.map((t) => t.destination),
+          transactions.map((t) => t.value),
+          ensureLeading0x(transactions.map((t) => trimLeading0x(t.data)).join('')),
+          transactions.map((t) => trimLeading0x(t.data).length / 2)
+        )
+        logCost(`Requesting attestations takes ${requestTx.receipt.gasUsed} gas`)
+        totalCost += requestTx.receipt.gasUsed
+
+        // Add fake randomness so that issuers can be selected.
+        const requestBlockNumber = await web3.eth.getBlockNumber()
+        await random.addTestRandomness(requestBlockNumber + selectIssuersWaitBlocks, '0x1')
+
+        const select = {
+          value: 0,
+          destination: attestations.address,
+          // @ts-ignore
+          data: attestations.contract.methods.selectIssuers(identifier).encodeABI(),
+          nonce: 3,
+        }
+        const selectTx = await executeMetaTransaction(user, mtw, select)
+        logCost(`Selecting issuers takes ${selectTx.receipt.gasUsed} gas`)
+        totalCost += selectTx.receipt.gasUsed
+
+        const issuers = await attestations.getAttestationIssuers(identifier, mtw.address)
+        for (let i = 0; i < numAttestations; i++) {
+          const issuer = issuers[i]
+          ;[v, r, s] = await getVerificationCodeSignature(
+            mtw.address,
+            issuer,
+            identifier,
+            _accounts
+          )
+          const complete = {
+            value: 0,
+            destination: attestations.address,
+            // @ts-ignore
+            data: attestations.contract.methods.complete(identifier, v, r, s).encodeABI(),
+            nonce: i + 4,
+          }
+          const completeTx = await executeMetaTransaction(user, mtw, complete)
+          logCost(`Completing an attestation takes ${completeTx.receipt.gasUsed} gas`)
+          totalCost += completeTx.receipt.gasUsed
+        }
+        const [completed, total] = await attestations.getAttestationStats(identifier, user)
+        assertEqualBN(completed, numAttestations)
+        assertEqualBN(total, numAttestations)
+        logCost(`Onboarding a user takes ${totalCost} gas`)
+      })
+      // Deploying and initializing a proxy takes 695743 gas
+      // Setting the account takes 189023 gas
+      // Requesting attestations takes 219860 gas
+      // Selecting issuers takes 249338 gas
+      // Completing an attestation takes 144778 gas
+      // Completing an attestation takes 104194 gas
+      // Completing an attestation takes 104146 gas
+      // Onboarding a user takes 1707082 gas
+    })
+
+    describe('With EIP-1167', () => {
+      const Proxy: ProxyContract = artifacts.require('InitializableProxy')
+      const ProxyCloneFactory: ProxyCloneFactoryContract = artifacts.require('ProxyCloneFactory')
+      let proxyCloneFactory: ProxyCloneFactoryInstance
+      before(async () => {
+        const proxy: ProxyInstance = await Proxy.new()
```

Comment:
note that we need to initialize the master `Proxy` when we deploy it

Reply:
https://github.com/optionality/clone-factory#warnings

Reply:
We do, in the constructor. But also I don't think that's relevant to these tests